Mathlib/RingTheory/Valuation/RankOne.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright (c) 2024 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández
+-/
+import Mathlib.Data.Real.NNReal
+import Mathlib.RingTheory.Valuation.Basic
+
+/-!
+# Rank one valuations
+
+We define rank one valuations.
+
+## Main Definitions
+* `IsRankOne` : A valuation has rank one if it is nontrivial and its image is contained in `ℝ≥0`.
+
+## Tags
+
+valuation, rank one
+-/
+
+noncomputable section
+
+open Function Multiplicative
+
+open scoped NNReal
+
+variable {R : Type*} [Ring R] {Γ₀ : Type*} [LinearOrderedCommGroupWithZero Γ₀]
+
+namespace Valuation
+
+/-- A valuation has rank one if it is nontrivial and its image is contained in `ℝ≥0`. -/
+class IsRankOne (v : Valuation R Γ₀) where
+  /-- The inclusion morphism from `Γ₀` to `ℝ≥0`. -/
+  hom : Γ₀ →*₀ ℝ≥0
+  strictMono' : StrictMono hom
+  nontrivial' : ∃ r : R, v r ≠ 0 ∧ v r ≠ 1
+
+namespace IsRankOne
+
+variable (v : Valuation R Γ₀) [IsRankOne v]
+
+lemma strictMono : StrictMono (hom v) := strictMono'
+
+lemma nontrivial : ∃ r : R, v r ≠ 0 ∧ v r ≠ 1 := nontrivial'
+
+/-- If `v` is a rank one valuation and `x : Γ₀` has image `0` under `is_rank_one.hom v`, then
+  `x = 0`. -/
+theorem zero_of_hom_zero {x : Γ₀} (hx : IsRankOne.hom v x = 0) : x = 0 := by
+  have hx0 : 0 ≤ x := zero_le'
+  cases' le_iff_lt_or_eq.mp hx0 with h_lt h_eq
```

Comment:
I wasn't sure if `cases'` was still OK (I asked [here](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Is.20cases'.20deprecated.3F/near/433616450), sorry to hold things up).

Reply:
I can change it to cases.